src/Codeception/Lib/Console/Output.php

```diff
@@ -67,7 +67,11 @@ protected function clean(string $message): string
 
     public function debug(mixed $message): void
     {
-        $message = print_r($message, true);
+        if (!is_scalar($message)) {
+            $this->writeln("\n<debug>  DEBUG OUTPUT:</debug>");
+            dump($message);
+            return;
+        }
         $message = str_replace("\n", "\n  ", $message);
         $message = $this->clean($message);
```

Comment:
What? We have no unit tests for this method?

Cleanup code here could be related to result of `print_r`.

Early return breaks `waitForDebugOutput` logic below.

Reply:
Thanks, good point. I don't know if this change can be tested as dump() prints directly to stdout
but I can try to check for side-effects, yes

Reply:
The point of `waitForDebugOutput` is to print the first debug message on a new line.
I think that it should be moved to the top of this method and `\n` should be removed from DEBUG OUTPUT.

Also `waitForDebugOutput` can be made private, because there is no code using it from outside this class anymore.

Reply:
Old code below could only handle strings, because other types produced type errors.

I removed all that code and the output looks like thi:
![image](https://user-images.githubusercontent.com/395992/161310176-e4462a2c-bd46-45d0-819f-6d09c822ce18.png)

Reply:
Do you think that `DEBUG OUTPUT:` line is necessary?

Reply:
strings really deserve special handling:
![image](https://user-images.githubusercontent.com/395992/161313013-c7b868fa-5728-4c64-a5eb-681242514b58.png)

Reply:
I think there must be a separator.
It could be anything from `-------` to `DEBUG OUTPUT:` although the latter are words to -ignore- when reading.

Reply:
we could add the type of data being printed and its size maybe.

Reply:
![image](https://user-images.githubusercontent.com/395992/161314113-ca2c85f3-de2b-482b-b07a-6d767e6ec7ad.png)

Reply:
![image](https://user-images.githubusercontent.com/395992/161314305-326a0fc3-0997-4e3e-a967-417da0e18418.png)

Reply:
I think delimiters for multiline output still can be a good idea, or not? 
but I agree, we should remove DEBUG OUTPUT
and print scalar values without var dumper

Reply:
you can take over now, I won't make any more changes today.

Reply:
I removed `DEBUG OUTPUT` so the output looks like this:
![image](https://user-images.githubusercontent.com/395992/161594962-ee45b5d7-6a9c-4bee-a858-dae59ae02f45.png)

It isn't necessary to add delimiters to string output, because when Codeception passes a string, it usually comes with meaningful prefix or format, adding delimiters would make it looks worse.
https://github.com/search?p=2&q=org%3ACodeception+codecept_debug&type=Code